components/datagrid/HeaderRowOverlay/HeaderRowOverlay.js

```diff
@@ -0,0 +1,47 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Box from '../../box';
+import BulkActions from './BulkActions';
+import NumSelectedRows from './NumSelectedRows';
+import theme from './theme.css';
+
+class HeaderRowOverlay extends PureComponent {
+  render() {
+    const { children, className, numSelectedRows, numSelectedRowsLabel, ...others } = this.props;
+
+    const classNames = cx(theme['header-row-overlay'], className);
+
+    return (
+      numSelectedRows > 0 && (
```

Comment:
Can we not prevent the `HeaderRowOverlay` component from rendering at a higher level instead of doing the check in here?